From: Josef Bacik <jbacik@fusionio.com>
Date: Tue, 5 Nov 2013 11:11:40 -0500
Patch-mainline: 3.13
Git-commit: 4724b106b9b8e8b802ca6f6d8a2f74feb8a3c375
References: FATE#312888
Subject: [PATCH] Btrfs: don't BUG_ON() if we get an error walking
 backrefs

We can just return false for this so we stop doing the snapshot aware defrag
stuff.  Thanks,

Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: Chris Mason <chris.mason@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/inode.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -2190,7 +2190,8 @@ static noinline bool record_extent_backr
 						  old->extent_offset, fs_info,
 						  path, record_one_backref,
 						  old);
-		BUG_ON(ret < 0 && ret != -ENOENT);
+		if (ret < 0 && ret != -ENOENT)
+			return false;
 
 		/* no backref to be processed for this extent */
 		if (!old->count) {
